projects/core/src/cart/facade/cart.service.ts

```diff
@@ -223,4 +225,52 @@ export class CartService {
   private isLoggedIn(userId: string): boolean {
     return typeof userId !== 'undefined';
   }
+
+  // TODO: Remove once backend is updated
+  /**
+   * Temporary method to merge guest cart with user cart because of beackend limitation
+   * This is for an edge case
+   */
+  private guestCartMerge(): void {
+    let cartEntries;
+    this.getEntries()
+      .pipe(take(1))
+      .subscribe(entries => {
+        cartEntries = entries;
+      });
+
+    this.store.dispatch(
+      new CartActions.DeleteCart({
+        userId: ANONYMOUS_USERID,
+        cartId: this.cartData.cart.guid,
+      })
+    );
+
+    combineLatest([
+      this.store.select(CartSelectors.getCartContent),
+      this.store.select(CartSelectors.getCartLoading),
+      this.getLoaded(),
+    ])
+      .pipe(
+        filter(([, loading]) => !loading),
```

Comment:
Can we just call 'function addEntry', which check cart exist or not; if not created, it will create a new cart first. So, the logic is `if cart is not loading, then addEntry`.